manifests/files.pp

```diff
@@ -2,73 +2,77 @@
 #
 # This is a defined type to create a swap_file
 #
-# == Parameters
-# [*ensure*]
+# @param ensure
 #   Allows creation or removal of swapspace and the corresponding file.
-# [*swapfile*]
+#
+# @param swapfile
 #   Location of swapfile, defaults to /mnt
-# [*swapfilesize*]
+#
+# @param swapfilesize
 #   Size of the swapfile as a string (eg. 10 MB, 1.2 GB).
 #   Defaults to $::memorysize fact on the node
-# [*add_mount*]
+#
+# @param add_mount
 #   Add a mount to the swapfile so it persists on boot
-# [*options*]
+#
+# @param options
 #   Mount options for the swapfile
-# [*timeout*]
+#
+# @param timeout
 #   dd command exec timeout.
 #   Defaults to 300
-# [*cmd*]
+#
+# @param cmd
 #   What command is used to create the file, dd or fallocate. dd is better tested and safer but fallocate is significantly faster.
 #   Defaults to dd
 #
-# == Examples
+# @param resize_existing
+#   Boolean to choose if existing swap files should get resized.
 #
+# @param resize_margin
+#   Margin that is checked before resizing the swapfile.
+#
+# @param resize_verbose
+#   Boolean to choose if a notify message to explain the change should be added.
+#
+# @examples
 #   swap_file::files { 'default':
 #     ensure   => present,
 #     swapfile => '/mnt/swap.55',
 #   }
 #
-# == Authors
+# @author
 #    @petems - Peter Souter
 #
 define swap_file::files (
-  $ensure        = 'present',
-  $swapfile      = '/mnt/swap.1',
-  $swapfilesize  = $::memorysize,
-  $add_mount     = true,
-  $options       = 'defaults',
-  $timeout       = 300,
-  $cmd           = 'dd',
-  $resize_existing = false,
-  $resize_margin   = '50MB',
-  $resize_verbose  = false,
-)
-{
-  # Parameter validation
-  validate_legacy(String, 'validate_re', $ensure, ['^absent$', '^present$'])
-  validate_legacy(String, 'validate_string', $swapfile)
+  Enum['present', 'absent'] $ensure          = 'present',
+  Stdlib::Absolutepath      $swapfile        = '/mnt/swap.1',
+  String                    $swapfilesize    = $::facts['memory']['system']['total'],
+  Boolean                   $add_mount       = true,
+  String                    $options         = 'defaults',
+  Integer                   $timeout         = 300,
+  String                    $cmd             = 'dd',
+  Boolean                   $resize_existing = false,
+  String                    $resize_margin   = '50MB',
+  Boolean                   $resize_verbose  = false,
+) {
   $swapfilesize_mb = to_bytes($swapfilesize) / 1048576
-  validate_legacy(Boolean, 'validate_bool', $add_mount)
 
   if $ensure == 'present' {
-
-    if ($resize_existing and $::swapfile_sizes) {
-
-      if (is_hash($::swapfile_sizes)) {
-
-        if (has_key($::swapfile_sizes,$swapfile)) {
+    if ($resize_existing and $::facts['swapfile_sizes']) {
+      if (is_hash($::facts['swapfile_sizes'])) {
```

Comment:
Do we know why this logic is necessary? I mean handling it both as a hash and as as CSV-formatted.

Reply:
We/I could further refactor it. Should be easy to use something like:

```
if (has_key($facts['swapfile_sizes'],$swapfile)) {
  $actual_swapfile_size = $facts['swapfile_sizes'][$swapfile]
} else {
  $actual_swapfile_size = $swapfilesize
}
swap_file::resize { $swapfile:
  swapfile_path          => $swapfile,
  margin                 => $resize_margin,
  expected_swapfile_size => $swapfilesize,
  actual_swapfile_size   => $actual_swapfile_size,
  verbose                => $resize_verbose,
  before                 => Exec["Create swap file ${swapfile}"],
}
```

Please let me know if I should further refactor the module since we have forked it now.

Reply:
> Do we know why this logic is necessary? I mean handling it both as a hash and as as CSV-formatted.

My guess would be that `swapfile_size` is better but not always working correctly and `swapfile_size_csv` is the fallback.

Reply:
I was partly right. The CSV facts seems to be used for older Puppet versions:
https://github.com/petems/petems-swap_file/pull/55#issuecomment-213742650

Reply:
Let's leave it for now.

Reply:
Have added a comment regarding the old Puppets to the code.